### PR TITLE
539 activate merge mode on existing tables

### LINKED ISSUE #539: Activate Merge Mode on existing tables
**Is your feature request related to a problem? Please describe.**
When using merge mode on a historization action, a column dl_hash is added to your DataObject which is used to detect changes. 
This works fine when merge mode is activated for a new DataObject (that has never been written before). 
When activating it for a DataObject with existing data, schema evolution takes care to add the column dl_hash, but it's empty for all existing records. 

**Describe the solution you'd like**
First, check if the current solution works. It's likely, that all previous records (with empty dl_hash) get closed and a new record with dl_hash is created. If that's the case, we can discuss to leave it like that.
The cleaner solution would be to calculate dl_hash for all previous records on activation.

### PR DESCRIPTION
**Pull request by ThierryGirod**
### What changes are included in the pull request?
 
Fix when using merge mode on a historization action the added column dl_hash was NULL.

### Why are the changes needed?
View issue #539

### CODE REVIEW

**Review by zzeekk — APPROVED** [review]
LGTM.
Should we do this for Iceberg as well?
Is the PR still "draft" by intention?

**Comment by ThierryGirod** [comment]
@zzeekk yes its intentional, because I am currently working on it for iceberg and jdbc.

File: sdl-core/src/main/scala/io/smartdatalake/util/evolution/SchemaEvolution.scala [hunk 1]
```diff
@@ -173,17 +186,19 @@ object SchemaEvolution extends SmartDataLakeLogger {
       // this defines the ordering of the resulting DataFrame's
       val tgtCols = if (Environment.schemaEvolutionNewColumnsLast) {
         // new columns last
-        oldColsWithoutTechCols ++ newColumns(oldDf, newDf) ++ colsToIgnore
+        if (caseSensitiveComparison) oldColsWithoutTechCols ++ newColumns(oldDf, newDf) ++ colsToIgnore
+        else oldColsWithoutTechCols ++ newColumns(oldDf, newDf) ++ colsToIgnore.map(_.toLowerCase)
```
Comment (zzeekk) [inline sdl-core/src/main/scala/io/smartdatalake/util/evolution/SchemaEvolution.scala]:
```suggestion
        if (caseSensitiveComparison) oldColsWithoutTechCols ++ newColumns(oldDf, newDf) ++ (if (caseSensitiveComparison) colsToIgnore else colsToIgnore.map(_.toLowerCase))
```

File: sdl-core/src/main/scala/io/smartdatalake/util/evolution/SchemaEvolution.scala [hunk 2]
```diff
@@ -173,17 +186,19 @@ object SchemaEvolution extends SmartDataLakeLogger {
...
       } else {
         // deleted columns last
-        newColsWithoutTechCols ++ deletedColumns(oldDf, newDf) ++ colsToIgnore
+        if (caseSensitiveComparison) newColsWithoutTechCols ++ deletedColumns(oldDf, newDf) ++ colsToIgnore
+        else newColsWithoutTechCols ++ deletedColumns(oldDf, newDf) ++ colsToIgnore.map(_.toLowerCase)
```
Comment (zzeekk) [inline sdl-core/src/main/scala/io/smartdatalake/util/evolution/SchemaEvolution.scala]:
```suggestion
        if (caseSensitiveComparison) newColsWithoutTechCols ++ deletedColumns(oldDf, newDf) ++ (if (caseSensitiveComparison) colsToIgnore else colsToIgnore.map(_.toLowerCase))
```

**Review by zzeekk — APPROVED** [review]
Well done!
Please use "squash and merge" to merge the PR...

File: sdl-core/src/main/scala/io/smartdatalake/util/evolution/SchemaEvolution.scala (hunk 1 above)
Comment (ThierryGirod) [reply sdl-core/src/main/scala/io/smartdatalake/util/evolution/SchemaEvolution.scala]: Fixed in Commit cb32221f47394f53c4b06c5d6b2c277df14e2d91

File: sdl-core/src/main/scala/io/smartdatalake/util/evolution/SchemaEvolution.scala (hunk 2 above)
Comment (ThierryGirod) [reply sdl-core/src/main/scala/io/smartdatalake/util/evolution/SchemaEvolution.scala]: Fixed in commit cb32221f47394f53c4b06c5d6b2c277df14e2d91

**Pull request merged**